### DISCUSSION TITLE
service_role request do not return data

### DISCUSSION THREAD

**Discussion by bkniffler** [supabase/supabase#3963] [Questions]
When making requests from postman with `ApiKey: $SERVICE_ROLE` header set, I'm not getting data (empty array, status 200) for a table setup with this RLS policy:

```
ALTER TABLE profiles ENABLE ROW LEVEL SECURITY;
CREATE POLICY "profiles_sub_policy" ON profiles USING (auth.role() = 'service_role');
```

Using `authenticated` role in policy doesn't work either. With `anon` role in policy, it works.

I thought service_role is meant to bypass RLS?

**Comment by bnjmnt4n** [comment]
Did you also set the `Authorization: Bearer ${SERVICE_ROLE}` header? I think both might be required to get the proper behaviour. From my knowledge, `apikey` is used by Kong, while the `Authorization` header is used by PostgREST.

**Reply by bkniffler** [reply]
Interesting, but I'm confused. `service_role` is listed in PG roles, and on decoding the `apikey`, there is `{ role: 'service_role' }`, so I suppose(d) that this role would be applied. So basically I'd have to give an Authorization header additionally with `{ role: 'service_role' }` payload?

I had hoped that the service role api key would be sufficient to talk to supabase from my backend, without having to add an additional `Authorization`, which would require a supabase user etc.

**Reply by bnjmnt4n** [reply]
I'm not too sure about the exact details about this, but my understanding is that the `apikey` header is only used by the Kong gateway. Hence, the `apikey` must be one of either the anon key or the service role key to ensure that any request is routed to the correct internal endpoint (PostgREST or GoTrue or storage etc.).

Then, the `Authorization` token is used by PostgREST to extract the JWT and get its claims, which are exposed in the `auth.role()`, `auth.email()` and `auth.uid()` functions. I.e., which role is used via PostgREST only depends on `Authorization` header and not the `apikey` header.

> I had hoped that the service role api key would be sufficient to talk to supabase from my backend, without having to add an additional Authorization, which would require a supabase user etc.

You can do this just by sending the `Authorization: Bearer ${TOKEN}`, where `TOKEN` can be the service role key as well (doesn't have to be the JWT associated with a user).

**Reply by bkniffler** [reply]
Just a little heads up @bnjmnt4n (and everyone else interested), it works good if send as `{ apikey: SERVICE_KEY, authorization: 'Bearer SERVICE_KEY' }`. Seems indeed like the one thing that really is important is the `Authorization` header, not so much apikey. Thanks again for pointing this out @bnjmnt4n.
